packages/cli/src/commands/push-bh.ts

```diff
@@ -0,0 +1,146 @@
+import { green } from 'colorette';
+
+export type PushBhOptions = {
+  organization?: string;
+  project: string;
+  mountPath: string;
+
+  branch: string;
+  author: string;
+  message: string;
+  files: string[];
+
+  domain?: string;
+  config?: string;
+};
+
+type FileToUpload = { path: string };
+
+export async function handleBhPush(argv: PushBhOptions, config: Config) {
+  const startedAt = performance.now();
+  const { organization, project: projectId, mountPath } = argv;
+
+  const orgId = organization || config.organization;
+
+  if (!orgId) {
+    return exitWithError(
+      `No organization provided, please use --organization option or specify the 'organization' field in the config file.`
+    );
+  }
+
+  const domain = argv.domain || BlueHarvest.getDomain();
+
+  if (!domain) {
+    return exitWithError(
+      `No domain provided, please use --domain option or environment variable REDOCLY_DOMAIN.`
+    );
+  }
+
+  try {
+    const author = parseCommitAuthor(argv.author);
+    const apiKey = BlueHarvest.getApiKeys(domain);
+    const filesToUpload = collectFilesToPush(argv.files);
+
+    if (!filesToUpload.length) {
+      return printExecutionTime('push-bh', startedAt, `No files to upload`);
+    }
+
+    const client = new BlueHarvest.ApiClient(domain, apiKey);
+    const projectDefaultBranch = await client.remotes.getDefaultBranch(orgId, projectId);
+    const remote = await client.remotes.upsert(orgId, projectId, {
+      mountBranchName: projectDefaultBranch,
+      mountPath,
+    });
+
+    process.stdout.write(
+      `Uploading ${filesToUpload.length} ${pluralize('file', filesToUpload.length)}:\n\n`
+    );
+
+    const { branchName: filesBranch } = await client.remotes.push(
+      orgId,
+      projectId,
+      remote.id,
+      {
+        commit: {
+          message: argv.message,
+          author,
+          branchName: argv.branch,
+        },
+      },
+      filesToUpload.map((f) => ({ path: f.path, stream: fs.createReadStream(f.path) }))
+    );
+
+    filesToUpload.forEach((f) => {
+      process.stdout.write(green(`✓ ${path.relative(process.cwd(), f.path)}\n`));
+    });
+
+    printExecutionTime(
+      'push-bh',
+      startedAt,
+      `${pluralize(
+        'file',
+        filesToUpload.length
+      )} uploaded to organization ${orgId}, project ${projectId}, branch ${filesBranch}`
+    );
+  } catch (err) {
+    exitWithError(`✗ File upload failed. Reason: ${err.message}\n`);
+  }
+}
+
+function parseCommitAuthor(author: string): { name: string; email: string } {
+  // Author Name <author@email.com>
+  const reg = /^.+\s<[^<>]+>$/;
+
+  if (!reg.test(author)) {
+    throw new Error('Invalid author format. Use "Author Name <author@email.com>"');
+  }
+
+  const [name, email] = author.split('<');
+
+  return {
+    name: name.trim(),
+    email: email.replace('>', '').trim(),
+  };
+}
+
+function collectFilesToPush(files: string[]): FileToUpload[] {
```

Comment:
Do we preserve nested folder structure? There could be references based on paths.

Reply:
Yes, we preserve it relative to entry points
e.g `push ../../folder --mount-path /docs/remotes`

Lets say content of the `../../folder` is next:

```treeview
├── app
│   ├── index.html
├── openapi.yaml
└── some-ref.yaml
``` 

The `/docs/remotes` (in BlueHarvest) will have the same content

Reply:
👍